app/services/facets/entry.rb

```diff
@@ -0,0 +1,67 @@
+module Facets
+  class Entry
+    attr_accessor :name, :model, :helper, :extension,
```

Comment:
If all of these variables are `attr_accessors`, what do we gain with the type checking below? In the end I could call `Facets::Entry.last.helper = 'not a module'` and it wouldn't complain. That's probably fine actually.

I'm not sure what to think about argument checking in Ruby, but I don't know any option that could be clearly better. I would suggest for this class to not worry about types, and let the Facet implementations worry about whether `:tabs` is an array, or whatever. 

The reason I propose this is so that any Facet could literally use anything as `tabs`, `helper`, etc... and all the `rescues`, exceptions and so forth would fail or raise an exception where they are called, not this early. It'd be more flexible and doesn't force us to define a very strongly typed type of Facet::Entry.

Summing up my point is that the responsibility for making sure a Facet works with the arguments provided during Facet registration lies on the person writing the Facet, not the "Facet framework".

@domcleal @jlsherrill any ideas?